ci/playbooks/kuttl/deploy-deps.yaml

```diff
@@ -0,0 +1,164 @@
+          {% set ns = namespace(found=false) -%}
+          {% for host in hostvars.keys() -%}
+          {%   if host is match('^compute.*') -%}
+          {%     set ns.found = true -%}
+          {%   endif -%}
+          {% endfor -%}
+          {{ ns.found }}
+
+    - name: Ensure that the isolated net was configured for crc
+      ansible.builtin.assert:
+        that:
+          - crc_ci_bootstrap_networks_out is defined
+          - "'crc' in crc_ci_bootstrap_networks_out"
+          - "'default' in crc_ci_bootstrap_networks_out['crc']"
+
+    - name: Ensure we have needed bits for compute when needed
+      when:
+        - has_compute | bool
+      ansible.builtin.assert:
+        that:
+          - "'compute' in crc_ci_bootstrap_networks_out"
+          - "'default' in crc_ci_bootstrap_networks_out['compute']"
+
+    - name: Set facts for further usage within the framework
+      ansible.builtin.set_fact:
+        cifmw_edpm_prepare_extra_vars:
+          NNCP_INTERFACE: "{{ crc_ci_bootstrap_networks_out.crc.default.iface }}"
+          NETWORK_MTU: "{{ crc_ci_bootstrap_networks_out.crc.default.mtu }}"
+
+- hosts: "{{ cifmw_target_host | default('localhost') }}"
+  name: Deploy Openstack Operators
+  tasks:
+
+    - name: Use the locally built operators if any
+      ansible.builtin.set_fact:
+        _local_operators_indexes: >-
+          {{
+            _local_operators_indexes|default({}) |
+            combine({ item.key.split('-')[0]|upper+'_IMG':
+            cifmw_operator_build_output['operators'][item.key].image_catalog})
+          }}
+      loop: "{{ cifmw_operator_build_output['operators'] | dict2items }}"
+      when:
+        - cifmw_operator_build_output is defined
+        - "'operators' in cifmw_operator_build_output"
+
+    - name: Set install_yamls Makefile environment variables
+      vars:
+        operators_build_output: "{{ (cifmw_operator_build_output | default({'operators':{}})).operators }}"
+        _openstack_operator_images:
+          OPENSTACK_IMG: "{{ cifmw_operator_build_output.operators['openstack-operator']['image_catalog'] | default(omit) }}"
+          OPENSTACK_BUNDLE_IMG: "{{ cifmw_operator_build_output.operators['openstack-operator']['image_bundle'] | default(omit) }}"
+      ansible.builtin.set_fact:
+        cifmw_edpm_prepare_common_env: >-
+          {{
+            cifmw_install_yamls_environment |
+            combine({'PATH': cifmw_path}) |
+            combine(cifmw_edpm_prepare_extra_vars | default({}))
+          }}
+        cifmw_edpm_prepare_operators_build_output: "{{ operators_build_output }}"
+        cifmw_edpm_prepare_make_openstack_env: "{{ _local_operators_indexes | combine(_openstack_operator_images) }}"
+
+    - name: Deploy openstack Operators
+      block:
+        - name: detect if openstack operator is installed
+          environment:
+            KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+            PATH: "{{ cifmw_path | default(ansible_env.PATH) }}"
```

Comment:
Why not passing this environment vars as part of the block to avoid repeating the same thing 4 times?

Reply:
good catch, done